tools/README.md

```diff
@@ -0,0 +1,39 @@
+# Tools for Kiali maintainers
+
+This folder contains tools for maintainers of the Kiali project.
+
+## Mock graph data
+
+There's two cli tools to help with testing large graph topologies without needing to deploy the underlying resources.
+
+### generator
+
+The generator creates a json file with a mock `/api/namespaces/graph` response.
+
+Running the following command will create a json file with 50 apps (workloads + services). There is some randomness in the graph generation so running the command multiple times with the same options will yield different results.
+
+```bash
+go run tools/cmd/generate/main.go -apps 50 -box
+```
+
+For more usage information:
+
+```bash
+go run tools/cmd/generate/main.go --help
+```
+
+### proxy server
+
+The proxy server runs an http proxy that intercepts all `/api/namespaces/graph` calls and returns mock graph data. The rest of the api calls are passed through to the kiali api. To test the kiali ui using the proxy server, you should start the proxy server then, in the kiali ui's `package.json`, replace the `proxy` field with the address of the proxy server e.g. `"proxy": "https://localhost:10201"`. You do not need to run the generator before running the proxy. The proxy server will run the generator itself from memory by default.
+
+Running the following command will start the proxy server in https mode.
+
+```bash
+go run tools/cmd/proxy/main.go -apps 50 -box -https
```

Comment:
Somehow my review comment got lost, I'll type it again...

This doesn't work because `-kiali-url` is required.

nit: I expected the options to use `--` as opposed to `-`.  Maybe it's just me but usually abbreviations use `-`.

The `-box` option needs to be tweaked to work like the normal ui.  Namespace boxing is only done when there are multiple namespaces, and also only when the box has multiple elements.  Here is an example of an unecessary box:

![image](https://user-images.githubusercontent.com/2104052/152607971-fea9bde0-ff6d-469a-988b-bf81c7599bca.png)

But overall it's pretty cool.  Just using the defaults from the doc:

![image](https://user-images.githubusercontent.com/2104052/152608166-cd634ed6-eeb6-48c6-8eeb-5f9764896a81.png)

By the way, dagre is terrible when you have an ingress gateway like this, with a lot of different destinations, because it stacks all of the ingress destinations.  Concentric looks nice because there is zero traffic between namespaces.

Reply:
> The -box option needs to be tweaked to work like the normal ui. Namespace boxing is only done when there are multiple namespaces, and also only when the box has multiple elements. Here is an example of an unecessary box:

I wonder if I could've had the input to `Generate` be a `trafficMap` instead of a `cytoscape.Config` and been able to re-use the appenders...

Reply:
Boxing actually does not happen in the appenders.  It happens in Cytoscape.go, which is the code that converts the TrafficMap into the cy config.   The idea being that we could take the generic TrafficMap structure and potentially generate the config for various vendors, it's just the case that we have only one vendor.  The config generators are fairly autonomous, at that point the TrafficMap is complete, there are no calls to kbs, prom, etc.